pkg/sources/github/github.go

```diff
@@ -333,16 +333,27 @@ func (s *Source) Chunks(ctx context.Context, chunksChan chan *sources.Chunk) err
 }
 
 func (s *Source) scan(ctx context.Context, installationClient *github.Client, chunksChan chan *sources.Chunk) error {
-	scanned := 0
+	var scanned uint64
 
 	log.Debugf("Found %v total repos to scan", len(s.repos))
 	wg := sync.WaitGroup{}
-	errs := []error{}
-	var errsMut sync.Mutex
+	errs := make(chan error, 1)
+	reportErr := func(err error) {
+		// save the error if there's room, otherwise log and drop it
+		select {
+		case errs <- err:
+		default:
+			log.WithError(err).Warn("dropping error")
+		}
+	}
+
```

Comment:
Doesn't this only ever keep the first error, which I assume would be non fatal if this is called more than once?

Reply:
Yes, I believe that was the previous behavior as well? Or rather, the previous behavior saved all errors in a slice and only returned the first if there was one.

Reply:
Hmm the previous behavior may not have been correct then. What do you think?

Reply:
Hm.. the only advantage I see of returning all the errors is if we want to determine control flow from the specific errors or display them to the user. In its current state we log each error, so we still have the information in the logs.

This function is essentially returning whether 1 or more errors occurred, which I think is fine for the `Chunks` method.